crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs

```diff
@@ -245,26 +276,29 @@ impl CheckpointExecutor {
 
         pending.push_back(spawn_monitored_task!(async move {
```

Comment:
As mentioned above, can we not just add the `send` here? Then we can rely on the `highest_executed_checkpoint` watermark for crash recovery (and can do the same for state accumulator). 

This would required that 
1. Pruner is implemented with idempotence
2. Order doesn't matter for pruning